### PR TITLE
Remove remaining "when C++17" comments

### PR DESCRIPTION
**Pull request by codereport**
Small cleanup PR. Follow up to #7931

### CODE REVIEW

File: cpp/include/cudf/detail/aggregation/aggregation.cuh [hunk 1]
```diff
@@ -632,27 +632,17 @@ struct identity_initializer {
...
-  typename std::enable_if<!cudf::is_timestamp_t<T>::value, T>::type get_identity()
-  {
-    if (k == aggregation::ARGMAX)
-      return static_cast<T>(ARGMAX_SENTINEL);
-    else if (k == aggregation::ARGMIN)
-      return static_cast<T>(ARGMIN_SENTINEL);
-    else
-      // In C++17, we can use compile time if and not make this function SFINAE
-      return identity_from_operator<T, k>();
+    if (k == aggregation::ARGMAX || k == aggregation::ARGMIN) {
```
Comment (harrism) [inline cpp/include/cudf/detail/aggregation/aggregation.cuh:637]: Nice

File: cpp/include/cudf_test/column_utilities.hpp [hunk 2]
```diff
@@ -211,13 +211,12 @@ inline std::pair<thrust::host_vector<std::string>, std::vector<bitmask_type>> to
...
   std::vector<std::string> host_data;
   host_data.reserve(c.size());
-
-  // When C++17, replace this loop with std::adjacent_difference()
-  for (size_type idx = 0; idx < c.size(); ++idx) {
-    auto offset = h_offsets[idx];
-    auto length = h_offsets[idx + 1] - offset;
-    host_data.push_back(std::string(h_chars.data() + offset, length));
-  }
+  std::transform(
```
Comment (harrism) [inline cpp/include/cudf_test/column_utilities.hpp:214]: ah, adjacent_transform....

**Review by harrism — APPROVED** [review]

File: cpp/src/io/json/reader_impl.cu [hunk 3]
```diff
@@ -478,23 +478,21 @@ void reader::impl::set_data_types(rmm::cuda_stream_view stream)
...
-                     [&](auto const &ts) -> std::pair<std::string, data_type> {
-                       auto const &[col_name, type_str] = split_on_colon(ts);
-                       return {col_name, convert_string_to_dtype(type_str)};
-                     });
+      std::transform(
+        std::cbegin(dtype),
+        std::cend(dtype),
+        std::inserter(col_type_map, col_type_map.end()),
+        [&](auto const &ts) -> std::pair<std::string, data_type> {
+          auto const [col_name, type_str] = split_on_colon(ts);
```
Comment (robertmaynard) [inline cpp/src/io/json/reader_impl.cu:493]: What was the reason to drop from `auto const &[col_name, type_str]` ?

File: cpp/src/io/json/reader_impl.cu [hunk 4]
```diff
@@ -478,23 +478,21 @@ void reader::impl::set_data_types(rmm::cuda_stream_view stream)
...
-                     std::inserter(col_type_map, col_type_map.end()),
-                     [&](auto const &ts) -> std::pair<std::string, data_type> {
-                       auto const &[col_name, type_str] = split_on_colon(ts);
-                       return {col_name, convert_string_to_dtype(type_str)};
-                     });
+      std::transform(
+        std::cbegin(dtype),
+        std::cend(dtype),
+        std::inserter(col_type_map, col_type_map.end()),
+        [&](auto const &ts) -> std::pair<std::string, data_type> {
```
Comment (robertmaynard) [inline cpp/src/io/json/reader_impl.cu:492]: In the above `split_on_colon` we transformed the the lambda to have a deduced return type, and we didn't do likewise. Was this an oversight?

File: cpp/src/io/json/reader_impl.cu (hunk 3 above)
Comment (codereport) [reply cpp/src/io/json/reader_impl.cu:493]: `string_view` is basically a pointer + a length. My preference is to copy.

File: cpp/src/io/json/reader_impl.cu (hunk 4 above)
Comment (codereport) [reply cpp/src/io/json/reader_impl.cu:492]: It was indeed. Fixed.

**Review by robertmaynard — APPROVED** [review]

**Comment by codereport** [comment]
@gpucibot merge

**Pull request merged**